lib/_http_server.js

```diff
@@ -722,10 +722,12 @@ function onParserExecuteCommon(server, socket, parser, state, ret, d) {
 
       socket.readableFlowing = null;
 
+      socket.unshift(bodyHead);
+
       // Clear the requestTimeout after upgrading the connection.
       clearRequestTimeout(req);
 
-      server.emit(eventName, req, socket, bodyHead);
+      server.emit(eventName, req, socket, Buffer.alloc(0));
```

Comment:
I think it make sense to reuse a preallocated buffer instead of a new one for each request.